gprofiler/profilers/registry.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, List, Optional, Type
+
+
+class ProfilerArgument:
+    # It would have been better to replace with a dataclass, but since we want to support Python 3.6 this is the best
+    # alternative as we still need the get_dict method (which is not convenient to do with a namedtuple)
+    def __init__(
+        self,
+        name: str,
+        dest: str,
+        help: str,
+        default: Any = None,
+        action: Optional[str] = None,
+        choices: List[Any] = None,
+        type: Type = None,
+    ):
+        self.name = name
+        self.dest = dest
+        self.help = help
+        self.default = default
+        self.action = action
+        self.choices = choices
+        self.type = type
+
+    def get_dict(self) -> Dict[str, Any]:
+        return {key: value for key, value in self.__dict__.items() if value is not None}
+
+
+class ProfilerConfig:
+    def __init__(
+        self,
+        profiler_mode_help: str,
+        disablement_help: str,
+        possible_modes: List[str] = None,
+        default_mode: str = "enabled",
+        arguments: List[ProfilerArgument] = None,
+    ):
+        self.profiler_mode_help: str = profiler_mode_help
+        self.possible_modes: Optional[List[str]] = possible_modes
+        self.default_mode: str = default_mode
+        self.profiler_args: List[ProfilerArgument] = arguments if arguments is not None else []
+        self.disablement_help = disablement_help
+
+
+profilers_config: Dict[str, ProfilerConfig] = {}
+
+
+def register_profiler(
+    profiler_name: str,
+    possible_modes: Optional[List] = None,
+    default_mode="enabled",
```

Comment:
```suggestion
    default_mode: str,
```

no need for default + add typing